### ISSUE TITLE
examples/canon (and a few others) have illegal haddock, breaks build with ghc -haddock

### ISSUE THREAD

**Issue by jwaldmann** [music-suite/music-suite#109]
examples/canon.hs starts like this:
```

{-# LANGUAGE OverloadedStrings #-}

-- |
-- Simple canon on a familiar subject.
--
import Music.Prelude
```
Apparently, this is illegal markup: an initial `-- |` comment requires that `module Foo where` follows.

Illegal haddock breaks compilation with `ghc-options: -haddock`, which I have in my global `.cabal/config`

```
cabal -w/opt/ghc/ghc-8.10.7/bin/ghc build
...
[1 of 1] Compiling Main             ( examples/chords.hs, nothing )

examples/canon.hs:6:1: error: parse error on input ‘import’
  |
6 | import Music.Prelude
  | ^^^^^^
```
I find `ghc -haddock`  immensly useful, as it enables the `:doc` command in `ghci`, as in
```
*Music.Prelude.Standard> :doc undelay

 Moves a value backward in time. Equnitvalent to @'stretch' . 'negate'@.
```

The problem only appears in examples/{hello, part, canon, fugue1} and it can be fixed trivially by either
* introducing a module name (make haddock happy), 
* or removing the `|` from the comment (so haddock ignores it)

I can send a PR. What option do you want?

NB: the problem (ghc dying on illegal haddock) will go away with ghc-9.0, cf.  https://gitlab.haskell.org/ghc/ghc/-/issues/17544#note_274255

**Comment by hanshoglund** [comment]
Thanks for the report.

I believe this was fixed in #110. Closing this.

**Comment by hanshoglund** [comment]
Thanks for the report and the fix.

I believe this was fully fixed in #110, so I will close this ticket.
